-- a/drivers/net/phy/realtek_switch/rtl8370mb/rtk_switch.h	2024-12-08 14:38:00.949856152 +0000
+++ b/drivers/net/phy/realtek_switch/rtl8370mb/rtk_switch.h	2024-12-10 02:48:49.496617360 +0000
@@ -30,8 +30,10 @@
 #define RTK_API_LOCK()      pthread_mutex_lock(&api_mutex)
 #define RTK_API_UNLOCK()    pthread_mutex_unlock(&api_mutex)
 #else
-#define RTK_API_LOCK()
-#define RTK_API_UNLOCK()
+#include <linux/mutex.h>
+extern struct mutex rtk_api_lock;
+#define RTK_API_LOCK()		mutex_lock(&rtk_api_lock)
+#define RTK_API_UNLOCK()	mutex_unlock(&rtk_api_lock)
 #endif
 
 typedef enum init_state_e
diff -urN a/drivers/net/phy/realtek_switch/rtl8370mb_switch.c b/drivers/net/phy/realtek_switch/rtl8370mb_switch.c
--- a/drivers/net/phy/realtek_switch/rtl8370mb_switch.c	2024-12-08 14:38:00.956856227 +0000
+++ b/drivers/net/phy/realtek_switch/rtl8370mb_switch.c	2024-12-10 02:26:29.863221596 +0000
@@ -18,6 +18,7 @@
 #include <linux/delay.h>
 #include <linux/skbuff.h>
 #include <linux/switch.h>
+#include <linux/mutex.h>
 
 //include from rtl8367c dir
 #include  "rtk_switch.h"
@@ -31,6 +32,7 @@
 /* RTL8370MB_PHY_PORT_NUM + ext1(cpu port) */
 #define RTL8370MB_NUM_PORTS		9 
 #define RTL8370MB_NUM_VIDS		4096
+DEFINE_MUTEX(rtk_api_lock);
 
 struct rtl8370mb_priv {
 	struct switch_dev	swdev;
